### PR TITLE
Transportation emissions, 2002 - 2022 using territorial framework

### LINKED ISSUE #120: EPA emissions - EQUATES intermediary years
Folks at the EPA are going to make smaller zip files with just on-road emissions inventory data for each year between 2002 and 2019. 

The currently available data is only available in very large compressed files (over 50 GB when downloaded and unzipped). I processed the data for years 2002, 2005, 2008, 2011, 2014, 2017, 2018 and 2019.

Once the data are available, we will replace the interpolated data with the actual data. 

See [CMAS forum post](https://forum.cmascenter.org/t/nitrous-oxide-n2o-availability-in-equates-county-level/5199/10) for details.

### PR DESCRIPTION
**Pull request by eroten**
Major changes throughout transportation sector. 

- Transportation emissions
  - New complete emissions dataset from 2002 - 2022 from various EPA sources. 
  - `compile_county_emissions.R` script is updated and tested 
  - All data are compiled from SMOKE FF10 downloads.
  - EPA data downloads are logged and documented in _transportation/data-raw/epa/epa_downloads.xlsx and described in README_epa_downloads.html.
  - EPA NEI
    - NEI 2020 is used in the final inventory.
    - Scripts for pulling summary tables from EnviroFacts, reading in on-road regional summary tables, and reading in SMOKE FF10 data are included.
    - EnviroFacts API version pulls multiple sectors, counties, overall more data.
  - EQUATES
    - EQUATES contains on-road vehicle emissions from 2002 to 2019 with consistent vehicle and fuel types. All data were sourced from the CMAS Data Warehouse. No interpolation is necessary (closes #120)
  - Air Emissions Modeling
    - Air emissions modeling data come from individual modeling platform releases. These are used for years 2021 and 2022 at this time. 
  - Non-road data processing has begun, but is not finalized. 
  - Essential datasets and documents used for validation and documentation are included. 

Documentation
- Transportation documentation is now separated out into the discussed results, methods/data sources/validation framework
- Placeholders are used to mark where future validation/verification will take place in documentation
- README_epa_downloads.Rmd and subsequent HTML file, along with epa_downloads.xlsx document downloads from the EPA FTP site and the CMAS Data Warehouse.
- One email thread `_transportation/data-raw/epa/Re: Seeking guidance on consistent sccs.pdf` is saved for future use regarding source classification codes.
- For now, old documentation Quarto documents have been moved to _transportation/deprecated for easy access. Much of the code will be useful for future documentation.
  

Datasets and scripts
-` _meta/epa_us_ghgi.R` reads in the 2021 edition of the  Inventory of U.S. Greenhouse Gas Emissions and Sinks by State: 1990-2021 (EPA GHGI).
- Transportation data/methods now lives in `_transportation/data_methods_transportation.qmd`
- `_transportation/_transportation_RDG.R` compiled basic emissions data for the first deadline of the Regional Development Guide. See #124
- `_transportation_compile.R` takes the very large compiled EPA dataset and summarizes it down to the  emissions_year, data_source, geoid, county_name, vehicle_weight_label, and vehicle_fuel_label. This is also where non-road and aviation data will be brought in.
- Many transportation scripts, see _script_descriptions.csv for more detail.

New functions
- `read_smoke_ff10` reads in a SMOKE FF10 file from a given location, filters the data to only include MN and WI counties and certain pollutants, and saves the resulting data in a specified output location. Records metadata and original file names for better documentation. Lives in `_transportation/data-raw/_read_smoke_ff10.R`
- `listify()` takes in a vector and creates a formatted sentence fragment with commas and an "and" preceding the last item in the vector. Taken from a neat [blog post](https://rfortherestofus.com/2022/10/automating-sentences-with-r). This lives in `R/_quarto_helpers.R`
- `round_emissions_metric_tons_co2e()` takes in a numeric vector and returns a nicely formatted sentence stating CO~2~e emissions in million or thousand metric tons, depending on value. Lives in `R/_plotting_helpers()`

Other
- Reference to IPCC AR5 corrected to IPCC AR6 in supplementary_data.qmd
- Cleaned up references to, file names, data storage locations for the Inventory of U.S. Greenhouse Gas Emissions and Sinks by State: 1990-2021 (EPA GHGI). Some data processing lived in _transportation/ and waste, but now everything lives in _meta/.
- General clean-up of unused scripts
- Bibliography up to date via Zotero.
- .gitignore up to date
- Styling changes throughout
- `download_read_table()` now checks for an existing file in the given exdir before downloading from internet location. New parameter `force_download` bypasses this condition, so you can still make a fresh download if desired. 
- Transportation colors updated in cprg_colors.R
- Additional testing for compiled county emissions dataset
- plotting functions throw a warning if you attempt to plot more than one year on a single Emissions/County plot
- source classification codes for transportation only compiled in `_transportation/data-raw/epa_source_classification_codes.R`. See #109 

Future considerations
- We can integrate [Quarto variables](https://quarto.org/docs/authoring/variables.html) across all Quarto documents to specify what our "current year" emissions year is. This way, all sectors will use the same year designation ("2021", "2022") to filter the existing plots and datasets.

Like with previous PRs on dev-2005-baseline, the entire Quarto document doesn't knit neatly, but all chunks in the transportation sections run without errors. 

## Checklist

Please complete this checklist as a courtesy to the PR reviewer.  

Code and styling
* [x] All of the files/scripts I added are in the right place and named appropriately. See the [README](https://github.com/Metropolitan-Council/ghg-cprg/blob/main/README.md) for details.
* [x] I have not used `setwd()`
* [x] I have used `file.path(here::here(), "file_name"))` to source any scripts or read in data
* [x] I have not added any large datasets, unless absolutely necessary - Some of the EPA datasets are very large, but have been compressed down. Intermediary .RDS data are available on MS Teams/OneDrive
* [x] I have commented my code, particularly in hard to understand areas
* [x] I have added additional package dependencies as necessary with `renv::install()`
* [x] I have run `styler::style_dir(".", recursive = TRUE, filetype = c("R", "qmd"))`
* [x] Plots
  - [x] If plotly, use `source = "chunk-name"` in `plot_ly()`
  - [x] Use formatting with `councilR::plotly_layout()`?
* [x] Chunk formatting
  - [x] All chunks named
  - [x] All figure or table chunks have caption
  - [x] `out.width: "95%"`. If a specific height is needed, use pixels. and/or `out.height: "500px"`

Document editing
* [x] I have ensured that modified documents knit successfully from [`render_for_publication.R`](https://github.com/Metropolitan-Council/ghg-cprg/blob/main/R/render_for_publication.R)
* [x] I have fixed any missing citations, cross references, hyperlinks
* [x] I have reviewed my contributions for typos and misspellings.

GitHub and project management
* [x] I have identified and assigned at least one Reviewer to this PR
* [x] I have assigned myself to this PR
* [x] I have updated the status in the [GitHub Project](https://github.com/orgs/Metropolitan-Council/projects/8/views/1)

### CODE REVIEW

File: _transportation/data-raw/epa/Re: Seeking guidance on consistent sccs.pdf
Comment (pawilfahrt) [inline _transportation/data-raw/epa/Re: Seeking guidance on consistent sccs.pdf:1]: @eroten Windows doesn't like the colon in there - can you rename and push?

File: _meta/data-raw/epa_us_ghgi.R [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  ) %>%
+  pivot_longer(starts_with("y"),
+    names_to = "inventory_year",
+    values_to = "emission_grams"
+  ) %>%
+  mutate(
+    inventory_year = str_remove(inventory_year, "y"),
+    # reported in millions of metric tons
+    # so convert to metric tons then to grams
+    emission_grams = as.numeric(emission_grams) * 1000000 %>%
```
Comment (pawilfahrt) [inline _meta/data-raw/epa_us_ghgi.R:43]: Doesn't this lead right back to where we started?

File: _transportation/data-raw/epa_air_emissions_modeling_activity.R
Comment (pawilfahrt) [inline _transportation/data-raw/epa_air_emissions_modeling_activity.R:1]: Seems like this script is unfinished or abandoned?

Comment (pawilfahrt) [reply _transportation/data-raw/epa_air_emissions_modeling_activity.R:1]: Nevermind - needed to reference script_descriptions

File: _transportation/data-raw/epa_onroad_emissions_compile.R [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+  unique() %>%
+  left_join(scc_combine, by = "scc6") %>%
+  arrange(emissions_year)
+
+# summarize combined datasets -----
+epa_emissions_summary <- epa_emissions_combine %>%
+  group_by(emissions_year, geoid, county_name, data_source) %>%
+  summarize(
+    emissions_metric_tons_co2e = sum(emissions_metric_tons_co2e),
+    co2 = sum(co2, na.rm =T),
```
Comment (pawilfahrt) [inline _transportation/data-raw/epa_onroad_emissions_compile.R:219]: Please review the addition of na.rm in these next three combined files - it seemed advantageous to me to have summed gases even when data was missing

File: _transportation/data-raw/epa_source_classification_codes.R
Comment (pawilfahrt) [inline _transportation/data-raw/epa_source_classification_codes.R:1]: code and created rds file looks good but read_download_table fails due to missing pathing. Additional permission issues when troubleshooting. Might be a PC issue?

**Review by pawilfahrt — CHANGES_REQUESTED** [review]
Whew! Amazing work Liz. There are some smallish changes with a few comments that should be quick approves or reverts.

The one exception is that directly reading files from urls continues to cause problems on my machine. Since this can be worked around by downloading and then directing to a file folder (less elegant but functional) I think it's okay to proceed and troubleshoot elsewhere at a later date.

File: _transportation/data-raw/epa_source_classification_codes.R
Comment (eroten) [reply _transportation/data-raw/epa_source_classification_codes.R:1]: I just pushed a couple changes that should address this. Can you try re-running?

File: _meta/data-raw/epa_us_ghgi.R (hunk 1 above)
Comment (eroten) [reply _meta/data-raw/epa_us_ghgi.R:43]: Yes. I'm not sure where I was going with this, but it isn't used for anything else right now. Can we leave it as is?

File: _transportation/data-raw/epa_source_classification_codes.R
Comment (eroten) [reply _transportation/data-raw/epa_source_classification_codes.R:1]: Resolved in 929b760a19848a9b1e083e98489659854e73e25c

File: _transportation/data-raw/epa_onroad_emissions_compile.R (hunk 2 above)
Comment (eroten) [reply _transportation/data-raw/epa_onroad_emissions_compile.R]: Yes, this was more important when I was prepping the data to be interpolated.

**Pull request merged**